src/compas/geometry/average.py

```diff
@@ -58,6 +60,25 @@ def centroid_points(points):
     x, y, z = zip(*points)
     return sum(x) / p, sum(y) / p, sum(z) / p
 
+def weighted_centroid_points(points, weights):
+    """Compute the weighted centroid of a set of points.
+
+    Parameters
+    ----------
+    points : list
+        A list of point coordinates.
+    weights : list
+        A list of weight floats.
+
```

Comment:
are there no bounds on the weights?

Reply:
but the weights are not bound to an interval between 0 and +/- 1?

Reply:
no the weights have no bounds